src/views/SessionView/index.vue

```diff
@@ -143,12 +149,18 @@ export default {
 
     promise
       .then(sessionId => {
+        this.$store.dispatch("user/updateSession", { _id: sessionId });
         this.$socket.io.opts.transports = ["polling", "websocket"];
-        this.$socket.connect();
-        this.joinSession(sessionId);
+        if (this.$socket.connected) {
+          this.joinSession(sessionId);
+        } else {
+          this.$socket.connect();
+        }
```

Comment:
does this mean sometimes the socket connection is already established once it gets here?

Reply:
In most cases the user actually establishes a socket connection to the server when they log in. This is necessary at least for volunteers; on their dashboards, the list of unfulfilled sessions needs to be kept up to date. We have been calling `connect` in spite of these already-established connections, but I want to try to refrain from unnecessarily creating new socket connections--this is probably leading to some confusion and chaos.

Reply:
👍